openedx/frontend-app-course-authoring
# Use object permissions in Tagging frontend [FC-0036]

## Description

Uses the permissions added to the Tagging REST API by https://github.com/openedx/openedx-learning/pull/138 to decide what actions (e.g. import, export, edit, delete) to present to the current user when viewing Tagging-related content. 

## Supporting information

Part of https://github.com/openedx/modular-learning/issues/160

This PR is dependent on these PRs:

* https://github.com/openedx/openedx-learning/pull/138
* https://github.com/openedx/edx-platform/pull/34004

## Testing instructions

Setup:

1. Run edx-platform from the branch on https://github.com/openedx/edx-platform/pull/34004
2. Install the updated requirements from ^.
3. Create test taxonomies, tags, orgs, and courses with https://github.com/open-craft/taxonomy-sample-data/
4. Run this PR's branch for course authoring.

Testing Tag Drawer:

1. Login to Studio as a non-superuser with course authoring permissions.
2. Visit the Sample Taxonomy Course course-v1:SampleTaxonomyOrg1+STC1+2023_1
3. Navigate to a Unit
4. Click "Manage Tags", either from the sidebar or the unit's menu
5. Ensure that the tag drawer opens with:
   * All the taxonomies the user is allowed to view
   * "Add tags" buttons for the listed taxonomies
   * Any selected object tags shown.
6. Add a bunch of tags at various hierarchy levels, and ensure they add as expected.
7. Delete some tags and ensure they delete as expected.

Testing Taxonomy Manager:

1. Login to Studio as a non-superuser with course authoring permissions.
   (Superusers have permission to do anything, and so aren't a good test of this change.)
2. Visit the Taxonomies list (http://localhost:2001/taxonomies)
3. Check that the "Import" button is enabled (new taxonomies can be added)
4. For each taxonomy listed, check that:
   * All taxonomies show "Export" as a menu option
   * Non-system taxonomies show "Re-import", "Manage organizations", and "Delete" as menu options
   * System taxonomies do not show "Re-import" or "Delete" as menu options

## Deadline

None

Private ref: FAL-3577
